autrainer/core/scripts/preprocess_script.py

```diff
@@ -118,110 +117,36 @@ def _preprocess_datasets(self) -> None:
-                    autrainer.instantiate_shorthand(t)
-                    for t in preprocess["pipeline"]
-                ]
-            )
-
-            file_count = 0
-            for file_path in chunk:
-                in_path = Path(dataset["path"], "default", file_path)
-                out_path = Path(
-                    dataset["path"],
-                    dataset["features_subdir"],
-                    file_path,
-                ).with_suffix("." + dataset["file_type"])
-                data = file_handler.load(in_path)
-                data = pipeline(data, 0)
-                output_file_handler.save(out_path, data)
-                del data
-                file_count += 1
-                if file_count % self.update_frequency == 0:
-                    with lock:
-                        pbar.update(self.update_frequency)
-            with lock:
-                pbar.update(file_count % self.update_frequency)
-
         print("Preprocessing datasets...")
         for (name, dataset), preprocess in zip(
             self.datasets.items(), self.preprocessing.values()
         ):
             print(f" - {name}")
-            # swap dataset handler with preprocessing handler
-            features_subdir = dataset["features_subdir"]
+            if preprocess is None:
+                print("No preprocessing specified. Skipping...")
+                continue
+            # set dataset file handler as output handler
+            # as dataset is already configured
+            # to work with the output of the preprocessing script
             output_file_handler = autrainer.instantiate_shorthand(
                 config=dataset["file_handler"],
                 instance_of=AbstractFileHandler,
             )
             output_file_type = dataset["file_type"]
-            # set features_subdir to None (defaults to audio)
-            # so that iteration takes place over raw data
-            dataset["features_subdir"] = None
+            
+            # override dataset file handling to work with raw audio
             dataset["file_handler"] = preprocess["file_handler"]
-            dataset["file_type"] = preprocess["file_type"]
-            if preprocess is None:
-                continue
+            # None allows dataset to work with all audio files
+            dataset["file_type"] = None
+            dataset["seed"] = 0  # ignored
+            dataset["batch_size"] = 8  # ignored
             data = autrainer.instantiate_shorthand(dataset)
-            loader = torch.utils.data.DataLoader(
-                torch.utils.data.ConcatDataset(
-                    data.train_dataset,
-                    data.dev_dataset,
-                    data.test_dataset
-                ),
-                shuffle=False,
-                num_workers=self.num_workers,
-                batch_size=1  #TODO: can we do it batched?
-            )
+            # manually disable dataset transforms
+            data.train_transform = None
```

Comment:
I think at this point you could handle the pipeline in a batched fashion (https://github.com/autrainer/autrainer/pull/54#discussion_r1824249027) by just setting `data.train_transform = pipeline ` (pipeline below), and then just straight up save `data[0]` from the loader.

Reply:
Do you foresee any unexpected errors when mapping the batched features to their original filenames using the returned index?

Reply:
No this should be fine and also could be avoided by not shuffling the loader at all.

Reply:
At second thought, this won't work unless we add `RandomCrop` since most of the time we have variable length sequences..

Reply:
Yeah we missed that since the features can have any size here!